fuzzers/libafl-fuzz/src/afl_stats.rs

```diff
@@ -0,0 +1,652 @@
+    saved_crashes: u64,
+    /// number of unique hangs encountered
+    saved_hangs: u64,
+    /// seconds since the last find was found
+    last_find: Duration,
+    /// seconds since the last crash was found
+    last_crash: Duration,
+    /// seconds since the last hang was found
+    last_hang: Duration,
+    /// execs since the last crash was found
+    execs_since_crash: u64,
+    /// the -t command line value
+    exec_timeout: u64,
+    /// real time of the slowest execution in ms
+    slowest_exec_ms: u128,
+    /// max rss usage reached during fuzzing in MB
+    peak_rss_mb: i64,
+    /// TODO
+    cpu_affinity: i64,
+    /// how many edges have been found
+    edges_found: u64,
+    /// TODO:
+    total_edges: u64,
+    /// how many edges are non-deterministic
+    var_byte_count: usize,
+    /// TODO:
+    havoc_expansion: usize,
+    /// TODO:
+    auto_dict_entries: usize,
+    /// TODO:
+    testcache_size: usize,
+    /// TODO:
+    testcache_count: usize,
+    /// TODO:
+    testcache_evict: usize,
+    /// banner text (e.g., the target name)
+    afl_banner: &'a Cow<'static, str>,
+    /// the version of AFL++ used
+    afl_version: &'a Cow<'static, str>,
+    /// default, persistent, qemu, unicorn, non-instrumented
+    target_mode: &'a Cow<'static, str>,
+    /// full command line used for the fuzzing session
+    command_line: &'a str,
+}
+
+#[derive(Debug, Clone)]
+pub struct AFLPlotData<'a> {
+    relative_time: &'a u64,
+    cycles_done: &'a u64,
+    cur_item: &'a usize,
+    corpus_count: &'a usize,
+    pending_total: &'a usize,
+    pending_favs: &'a usize,
+    /// Note: renamed `map_size` -> `total_edges` for consistency with `fuzzer_stats`
+    total_edges: &'a u64,
+    saved_crashes: &'a u64,
+    saved_hangs: &'a u64,
+    max_depth: &'a u64,
+    execs_per_sec: &'a u64,
+    /// Note: renamed `total_execs` -> `execs_done` for consistency with `fuzzer_stats`
+    execs_done: &'a u64,
+    edges_found: &'a u64,
+}
+
+impl<E, EM, Z> UsesState for AflStatsStage<E, EM, Z>
+where
+    E: UsesState,
+    EM: EventFirer<State = E::State>,
+    Z: UsesState<State = E::State>,
+{
+    type State = E::State;
+}
+
+impl<E, EM, Z> Stage<E, EM, Z> for AflStatsStage<E, EM, Z>
+where
+    E: UsesState,
+    EM: EventFirer<State = E::State>,
+    Z: UsesState<State = E::State>,
+    E::State: HasImported
+        + HasCorpus
+        + HasMetadata
+        + HasStartTime
+        + HasExecutions
+        + HasNamedMetadata
+        + Stoppable,
+{
+    fn perform(
+        &mut self,
+        _fuzzer: &mut Z,
+        _executor: &mut E,
+        state: &mut E::State,
+        _manager: &mut EM,
+    ) -> Result<(), Error> {
+        let Some(corpus_idx) = state.current_corpus_id()? else {
+            return Err(Error::illegal_state(
+                "state is not currently processing a corpus index",
+            ));
+        };
+        let testcase = state.corpus().get(corpus_idx)?.clone();
```

Comment:
You can use `state.current_testcase`

Reply:
We need the corpus_id down the line anyways so (for ``cur_item``) so I would keep it even if there's a bit of duplication